src/extension/alterschema/linter/else_false.h

```diff
@@ -0,0 +1,36 @@
+class ElseFalse final : public SchemaTransformRule {
+public:
+  ElseFalse()
+      : SchemaTransformRule{"else_false",
+                            "`if: S, else: false` collapses to just `S`"} {};
+
+  [[nodiscard]] auto
+  condition(const JSON &schema, const JSON &, const Vocabularies &vocabularies,
+            const SchemaFrame &, const SchemaFrame::Location &,
+            const SchemaWalker &, const SchemaResolver &) const
+      -> SchemaTransformRule::Result override {
+    return contains_any(
+               vocabularies,
+               {"https://json-schema.org/draft/2020-12/vocab/applicator",
+                "https://json-schema.org/draft/2019-09/vocab/applicator",
+                "http://json-schema.org/draft-07/schema#"}) &&
+           schema.is_object() && schema.defines("if") &&
```

Comment:
Should we also check for `then` being missing or being `true`? Otherwise the semantics could break. For example, what if `then` is `false` or `then` is `not: {}` (or any impossible schema)?